gpytorch/lazy/kronecker_sum_lazy_tensor.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+from typing import Optional
+
+from .sum_lazy_tensor import SumLazyTensor
+from .kronecker_product_lazy_tensor import KroneckerProductLazyTensor
+from .. import settings
+from ..utils.memoize import cached
+
+class KroneckerSumLazyTensor(SumLazyTensor):
+    r"""
+    Returns the sum of two Kronecker product lazy tensors. Solves and log-determinants
+    are computed using the eigen-decomposition of the right lazy tensor; that is,
+    (A \kron B + C \kron D) = (C^{1/2} \kron D^{1/2})
+        (C^{-1/2}AC^{-1/2} \kron D^{-1/2} B D^{-1/2} + I_|C| \kron I_|D|)(C^{1/2} \kron D^{1/2})^{T}
+    where .^{1/2} = Q_.\Lambda_.^{1/2} (e.g. an eigen-decomposition.)
+
+    This formulation admits efficient solves and log determinants.
+
+    The original reference is [https://papers.nips.cc/paper/2013/file/59c33016884a62116be975a9bb8257e3-Paper.pdf].
+
+    Args:
+        :`lazy_tensors`: List of lazy tensors
+    """
+
+    @property
```

Comment:
should this be cached or is this causing issues?

Reply:
I guess I'm not sure whether caching this here would have any major perf implications

Reply:
It is a root and several dense matmuls. It's mostly cached to ensure that the same root decomposition is used throughout.